### ISSUE TITLE
logger.end_task() empty values results in error

### ISSUE THREAD

**Issue by sameerkhnl** [Continvvm/continuum#262]
After using` loggger.end_task()`, it creates empty arrays after the last task ` [{'predictions': array([], dtype=float64),
     'targets': array([], dtype=float64),
     'task_ids': array([], dtype=float64)}]]},`

Because of this, logger.accuracy, logger.online_accuracy, logger.average_incremental_accuracy all result in errors. 
Here is my use of logger:

```
print('\n[Phase 3] : Model training')
logger = Logger(list_keywords=["performance"], list_subsets=["train", "test"])
elapsed_time = 0
for tid,taskset in enumerate(scenario_train):
    print(f'\n| Training on task {tid+1}: ')
    test_taskset = scenario_test[:tid + 1]
    test_loader = DataLoader(test_taskset, batch_size=batch_size,shuffle=False,num_workers=num_workers)
        
    for epoch in range(start_epoch,start_epoch + num_epochs):
        train_loader = DataLoader(taskset, batch_size=batch_size, shuffle=True, num_workers=num_workers, drop_last=True)
        start_time = time.time()
        total_batches = len(taskset) // batch_size
        train(epoch, net, train_loader, total_batches)

        epoch_time = time.time() - start_time
        elapsed_time += epoch_time
        print(f'| Elapsed time : {utils.get_hms(elapsed_time)}')        

    train_loader_eval = DataLoader(taskset, batch_size=batch_size, shuffle=False, num_workers=num_workers)
    eval(net, train_loader_eval, logger, 'train')  #logger subset

    eval(net, test_loader, logger, 'test') #logger subset

    logger.end_task()  

```

**Comment by arthurdouillard** [comment]
Right now the usage (not the best I agree) is that you finish the task, log whaterever you want, and finally call the `end_task` method.

You can see how I did in this project: https://github.com/arthurdouillard/dytox/blob/6c9f7d1cd4fdebe03ec9e2abba55b34cab5d1979/main.py#L669 (I log on disk metrics in the function just before).

**Comment by TLESORT** [comment]
I believe this issue is solved, so I will close it.
Feel free to reopen it if needed.
